group_project_v2/stage/review.py

```diff
@@ -27,11 +30,28 @@ class ReviewBaseStage(BaseGroupActivityStage):
     js_file = "public/js/stages/review_stage.js"
     js_init = "GroupProjectReviewStage"
 
+    review_item_key = None
+
     STAGE_ACTION = _(u"save feedback")
     FEEDBACK_SAVED_MESSAGE = _(u'Thanks for your feedback.')
 
     TA_GRADING_NOT_ALLOWED = _(u"TA grading is not allowed for this stage")
 
+    # (has_some, has_all) -> ReviewState. have_all = True and have_some = False is obviously an error
+    REVIEW_STATE_CONDITIONS = {
+        (True, True): ReviewState.COMPLETED,
+        (True, False): ReviewState.INCOMPLETE,
+        (False, False): ReviewState.NOT_STARTED
+    }
```

Comment:
Optional: It could arguably be easier to understand the code if you move this constant to be inside `_calculate_review_status()` which is the only place where it is used. Same for the next one, `STAGE_STATE_REVIEW_STATE_MAPPING`.